analyzer/src/main/kotlin/managers/Cargo.kt

```diff
@@ -214,6 +170,59 @@ class Cargo(
             packageDir.startsWith(analysisRoot)
         } ?: false
 
+    private fun buildDependencyTree(
+        name: String,
+        version: String,
+        packages: Map<String, Package>,
+        metadata: JsonNode
+    ): PackageReference {
+        val node = metadata["packages"].single {
+            it["name"].textValue() == name && it["version"].textValue() == version
+        }
+        val dependencies = node["dependencies"]
+            .filter {
+                // Filter dev and build dependencies, because they are not transitive.
+                val kind = it["kind"].textValueOrEmpty()
+                kind != "dev" && kind != "build"
+            }
+            .mapNotNull {
+                // TODO: Handle renamed dependencies here, see:
+                //       https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#renaming-dependencies-in-cargotoml
+                val dependencyName = it["name"].textValue()
+                val dependencyVersion = getResolvedVersion(name, version, dependencyName, metadata)
+                if (dependencyVersion == null) {
+                    null
+                } else {
+                    buildDependencyTree(dependencyName, dependencyVersion, packages, metadata)
+                }
+            }
+            .toSortedSet()
+
+        val id = extractCargoId(node)
+        val pkg = packages.getValue(id)
+        val linkage = if (isProjectDependency(id)) PackageLinkage.PROJECT_STATIC else PackageLinkage.STATIC
+        return pkg.toReference(linkage, dependencies)
+    }
+
+    private fun getResolvedVersion(
+        parentName: String,
+        parentVersion: String,
+        dependencyName: String,
+        metadata: JsonNode
+    ): String? {
+        val node = metadata["resolve"]["nodes"].single {
+            it["id"].textValue().startsWith("$parentName $parentVersion")
+        }
+
+        // This is null if the dependency is optional and the feature was not enabled. In this case the version was not
+        // resolved and the dependency should not appear in the dependency tree.
+        val dependency = node["dependencies"].find {
+            it.textValue().startsWith(dependencyName)
+        }
+
+        return dependency?.textValue()?.split(" ")?.get(1)
```

Comment:
Nit: Could use the `Char`-version of `split(' ')` here.

Reply:
Done.